core/src/main/java/bio/terra/pearl/core/service/publishing/PortalPublishingService.java

```diff
@@ -142,4 +150,44 @@ protected void applyChangesToNotificationConfigs(PortalEnvironment destEnv, List
         }
     }
 
+    protected void applyChangesToParticipantDashboardAlerts(PortalEnvironment destEnv, List<ParticipantDashboardAlertChange> changes) throws Exception {
+        for(ParticipantDashboardAlertChange change : changes) {
+            Optional<ParticipantDashboardAlert> destAlert = portalDashboardConfigService.findByPortalEnvIdAndTrigger(destEnv.getId(), change.trigger());
+            if(destAlert.isEmpty()) {
+                // The alert doesn't exist in the dest env yet, so default all the required fields before
+                // applying the changes from the change list
+                ParticipantDashboardAlert newAlert = ParticipantDashboardAlert.builder()
+                        .portalEnvironmentId(destEnv.getId())
+                        .type(AlertType.primary)
+                        .title("")
+                        .detail("")
+                        .portalEnvironmentId(destEnv.getId())
+                        .trigger(change.trigger())
+                        .build();
+
+                applyAlertChanges(newAlert, change.changes());
+                portalDashboardConfigService.create(newAlert);
+            } else {
+                ParticipantDashboardAlert alert = destAlert.get();
+                applyAlertChanges(alert, change.changes());
+                portalDashboardConfigService.update(alert);
+            }
+        }
+    }
+
+    protected void applyAlertChanges(ParticipantDashboardAlert alert, List<ConfigChange> changes) {
+        try {
+            for (ConfigChange alertChange : changes) {
+                //The 'type' property is an enum, so we need to handle it differently than the other properties
```

Comment:
Ah, great catch.  I just realized this bug impacts notification configs too -- if you attempt to change the event type, it will error.  

Instead of hardcoding this, let's make a new utility method "setPropertyEnumSafe" that uses reflection to detect whether the target property is an enum, and if so, convert the string value to an enum before invoking PropertyUtils.setProperty.  

I believe you can get the target class by using `PropertyUtils.getPropertyDescriptor(alert.getClass(), alertChange.propertyName).getPropertyType()`

Then you can upgrade `PublishingUtils.applyChangesToVersionedConfig` to use that method